coderd/workspaces_test.go

```diff
@@ -3659,58 +3687,82 @@ func TestWorkspaceTimings(t *testing.T) {
 		return dbgen.ProvisionerJobTimings(t, db, insertParams)
 	}
 
+	makeAgentScriptTimings := func(scriptID uuid.UUID, count int) []database.WorkspaceAgentScriptTiming {
+		newTimings := make([]database.InsertWorkspaceAgentScriptTimingsParams, count)
+		now := time.Now()
+		for i := range count {
+			startedAt := now.Add(-time.Hour + time.Duration(i)*time.Minute)
+			endedAt := startedAt.Add(time.Minute)
+			newTimings[i] = database.InsertWorkspaceAgentScriptTimingsParams{
+				StartedAt: startedAt,
+				EndedAt:   endedAt,
+				Stage:     database.WorkspaceAgentScriptTimingStageStart,
+				ScriptID:  scriptID,
+				ExitCode:  0,
+				Status:    database.WorkspaceAgentScriptTimingStatusOk,
+			}
+		}
+
+		timings := make([]database.WorkspaceAgentScriptTiming, 0)
+		for _, newTiming := range newTimings {
+			timing := dbgen.WorkspaceAgentScriptTiming(t, db, newTiming)
+			timings = append(timings, timing)
+		}
+
+		return timings
+	}
+
 	// Given
 	testCases := []struct {
-		name               string
 		provisionerTimings int
-		workspace          workspaceWithBuild
-		error              bool
+		agentScriptTimings int
```

Comment:
Removing the names of the tests and the comments has degraded the quality of these tests, I feel.
It's not clear to me what the intention is for testing for different number of timings.
Please break these up into clearly named scenarios, and make your intent clear.

Reply:
Makes sense. I based my decision on [this video](https://youtu.be/X4rxi9jStLo?t=364) that you shared with me some time ago about Go tests, so I thought it was a Go community convention. I'm going to add the names back.